projects/core/src/auth/facade/auth.service.ts

```diff
@@ -28,6 +28,26 @@ export class AuthService {
     );
   }
 
+  /**
+   * This function provides the userId the OCC calls should use, depending
+   * on wether there is an active storefront session or not.
+   *
+   * It returns the userId of the current storefront user or 'anonymous'
+   * in the case there are no signed in user in the storefront.
+   *
+   * The user id of a regular customer session is 'current'.  In the case of an
+   * asm customer emulation session, the userId will be the customerId.
+   */
+  getOccUserId(): Observable<string> {
+    return this.getUserToken().pipe(
+      map(userToken =>
+        Object.keys(userToken).length !== 0
```

Comment:
Is there a world somewhere where userToken could return null or undefined?

Reply:
I agree it's best to make this more robust.
I made the code resilient to undefined/null token or undefined/null userId property.